code/modules/credits/credits.dm

```diff
@@ -0,0 +1,190 @@
+		possible_titles += "A VERY [pick("CORPORATE", "NANOTRASEN", "FLEET", "CLOWN", "SYNDICATE", "EXPEDITIONARY", "EXPLORATORY", "PLASMA", "MARTIAN", "UNITED")] [pick("CHRISTMAS", "EASTER", "HOLIDAY", "WEEKEND", "THURSDAY", "VACATION")]"
+		possible_titles += "[pick("GUNS, GUNS EVERYWHERE", "THE LITTLEST PRIMALIS", "WHAT HAPPENS WHEN YOU MIX MAINTENANCE DRONES AND COMMERCIAL-GRADE PACKING FOAM", "ATTACK! ATTACK! ATTACK!", "HOLY SHIT A BOMB", "THE LEGEND OF THE ALIEN ARTIFACT: PART [pick("I","II","III","IV","V","VI","VII","VIII","IX", "X", "C","M","L")]")]"
+		possible_titles += "[pick("SPACE", "TERRIFYING", "DRAGON", "WARLOCK", "LAUNDRY", "GUN", "ADVERTISING", "DOG", "CARBON MONOXIDE", "NINJA", "WIZARD", "SOCRATIC", "JUVENILE DELIQUENCY", "POLITICALLY MOTIVATED", "RADTACULAR SICKNASTY")] [pick("QUEST", "FORCE", "ADVENTURE")]"
+		possible_titles += "[pick("THE DAY [pick("NANOTRASEN", "THE SYNDICATE", "ARMADYNE CORPORATION", "THE CLOWN",)] STOOD STILL", "HUNT FOR THE GREEN WEENIE", "ALIEN VS VENDOMAT", "SPACE TRACK")]"
+		titles += "<center><h1>EPISODE [rand(1,1000)]<br>[pick(possible_titles)]<h1></h1></h1></center>"
+	else
+		titles += "<center><h1>EPISODE [rand(1,1000)]<br>[GLOB.end_credits_title]<h1></h1></h1></center>"
+
+	// The Living
+	var/list/minds = list()
+	for(var/mob/living/carbon/human/H in GLOB.human_list)
+		if(H.mind)
+			minds += H.mind
+
+	for(var/datum/mind/mind in minds)
+		if(!mind)
+			continue
+		if(!cast.len && !chunksize)
+			chunk += "CREW:"
+		var/jobtitle = mind?.assigned_role || "No title"
+		var/used_name = mind?.current?.name
+		var/antag_string
+		for(var/datum/antagonist/antagonist as anything in mind?.antag_datums)
+			antag_string ? (antag_string += ", ") : (antag_string += "...")
+			antag_string += "[antagonist?.name]"
+		chunk += "[used_name] as the [mind?.antag_datums ? "[antag_string] and [jobtitle]" : jobtitle]"
+		chunksize++
+
+		if(chunksize > 2)
+			cast += "<center>[jointext(chunk,"<br>")]</center>"
+			chunk.Cut()
+			chunksize = 0
+	if(chunk.len)
+		cast += "<center>[jointext(chunk,"<br>")]</center>"
+
+	titles += cast
+
+	// The Dead
+	var/list/corpses = list()
+	var/list/monkies = list()
+	for(var/mob/living/carbon/human/H in GLOB.dead_mob_list)
+		if(H.timeofdeath < 5 MINUTES) // no prespawned corpses
+			continue
+		if(ismonkeybasic(H))
+			monkies[H?.name] += 1
+		else if(H?.real_name in GLOB.crew_list) // Only crew deaths
+			corpses += H?.real_name
+	if(corpses.len)
+		titles += "<center>BASED ON REAL EVENTS<br>In memory of [english_list(corpses)].</center>"
+
+	// CKEYS
+	var/list/staff = list("PRODUCTION STAFF:")
+	var/static/list/staffjobs = list("Coffee Fetcher", "Cameraman", "Angry Yeller", "Chair Operator", "Choreographer", "Historical Consultant", "Costume Designer", "Chief Editor", "Executive Assistant")
+	var/list/goodboys = list()
+	for(var/client/C)
+		if(!C?.holder)
+			continue
+		if(C?.holder)
```

Comment:
This doesn't respect hidden staff that are using big brother or stealth mode, I think there are 2 different types